### PR TITLE
Add POST DELETE beneficiary support

### PR DESCRIPTION
**Pull request by pkachelhoffer**
Luno recently added the POST and DELETE beneficiary api endpoints. Add in support for the new endpoints.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -948,6 +948,74 @@ func (cl *Client) ListBeneficiariesResponse(ctx context.Context, req *ListBenefi
...
+	// required: true
+	// example: 12345
+	ID string `json:"id,string" url:"id"`
+}
+
+// DeleteBeneficiary makes a call to DELETE /api/exchange/1/beneficiaries to delete a beneficiary.
+//
+// Permissions required: <code>Perm_W_Beneficiaries</code>
+func (cl *Client) DeleteBeneficiary(ctx context.Context, req *DeleteBeneficiaryRequest) error {
+	var res CreateBeneficiaryResponse
```
Comment (pierredup) [inline api.go:1011]: Should the response for this be `DeleteBeneficiaryResponse` instead?

File: api.go [hunk 2]
```diff
@@ -948,6 +948,74 @@ func (cl *Client) ListBeneficiariesResponse(ctx context.Context, req *ListBenefi
...
+	err := cl.do(ctx, "DELETE", "/api/1/beneficiaries/{id}", req, &res, true)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (pierredup) [inline api.go:1016]: Should the response (`res`) be returned here? Does this endpoint return anything in the response?

File: api.go [hunk 3]
```diff
@@ -948,6 +948,74 @@ func (cl *Client) ListBeneficiariesResponse(ctx context.Context, req *ListBenefi
...
+type DeleteBeneficiaryRequest struct {
+	// ID of the Beneficiary to delete.
+	//
+	// in: path
+	// required: true
+	// example: 12345
+	ID string `json:"id,string" url:"id"`
+}
+
+// DeleteBeneficiary makes a call to DELETE /api/exchange/1/beneficiaries to delete a beneficiary.
```
Comment (pierredup) [inline api.go:1007]:
```suggestion
// DeleteBeneficiary makes a call to DELETE /api/1/beneficiaries to delete a beneficiary.
```

File: api.go [hunk 4]
```diff
@@ -948,6 +948,74 @@ func (cl *Client) ListBeneficiariesResponse(ctx context.Context, req *ListBenefi
...
+	Recipient string `json:"bank_recipient" url:"bank_recipient"`
+}
+
+// CreateBeneficiaryResponse response object for creating beneficiaries
+type CreateBeneficiaryResponse struct {
+	// Identifier for the created beneficiary
+	ID string `json:"id"`
+}
+
+// CreateBeneficiary makes a call to POST /api/exchange/1/beneficiaries to create a new beneficiary.
```
Comment (pierredup) [inline api.go:983]:
```suggestion
// CreateBeneficiary makes a call to POST /api/1/beneficiaries to create a new beneficiary.
```

File: api.go (hunk 1 above)
Comment (pkachelhoffer) [reply api.go:1011]: Oops! Copy and paste strikes again. Fixing...

File: api.go (hunk 2 above)
Comment (pkachelhoffer) [reply api.go:1016]: This endpoint doesn't returning anything. If there is an error, the return error will be populated.

**Pull request closed without merging**